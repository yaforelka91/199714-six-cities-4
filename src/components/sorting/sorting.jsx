import React from 'react';
import {connect} from 'react-redux';
import {SortType} from '../../const.js';
import {sortingTypes} from '../../types/types.js';
import {getSorting} from '../../reducer/catalog/selectors.js';
import {ActionCreator} from '../../reducer/catalog/catalog.js';

const sortItems = [
  {
    type: SortType.POPULAR,
    text: `Popular`,
  },
  {
    type: SortType.TO_HIGH,
    text: `Price: low to high`,
  },
  {
    type: SortType.TO_LOW,
    text: `Price: high to low`,
  },
  {
    type: SortType.TOP_RATED,
    text: `Top rated first`,
  },
];

const Sorting = ({activeSorting, isOpen, onSortItemClick, onToggleMenu, onSelectMenu}) => {
  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      {`\u00A0`}
      <span className="places__sorting-type" tabIndex="0" onClick={onToggleMenu}>
        {sortItems.find((item) => item.type === activeSorting).text}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <ul className={`places__options places__options--custom ${isOpen ? `places__options--opened` : ``}`}>
        {
          sortItems.map((item)=>(
            <li
              key={item.type}
              className={`places__option ${item.type === activeSorting ? `places__option--active` : ``}`}
              tabIndex="0"
              onClick={() => {
                onSortItemClick(item.type);
                onSelectMenu();
              }}
            >
              {item.text}
            </li>
          ))
        }
      </ul>
      <select
        className="places__sorting-type"
        id="places-sorting"
        value={activeSorting}
        onChange={(evt) => {
          onSortItemClick(evt.target.value);
        }}
        style={{
          clip: `rect(1px, 1px, 1px, 1px)`,
          height: `1px`,
          border: 0,
          overflow: `hidden`,
          position: `absolute`,
          whiteSpace: `nowrap`,
          width: `1px`,
        }}
      >
        {
          sortItems.map((item)=>(
            <option
              key={item.type}
              className="places__option"
              value={item.type}
            >
              {item.text}
            </option>
          ))
        }
      </select>
    </form>
  );
};

Sorting.propTypes = sortingTypes;

const mapStateToProps = (state) => ({
  activeSorting: getSorting(state),
});

const mapDispatchToProps = (dispatch) => ({
  onSortItemClick(sortType) {
    dispatch(ActionCreator.changeSort(sortType));
  },
});

export {Sorting};
export default connect(mapStateToProps, mapDispatchToProps)(Sorting);
