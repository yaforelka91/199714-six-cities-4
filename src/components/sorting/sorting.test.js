import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {Sorting} from './sorting.jsx';

const mockStore = configureStore([]);

describe(`SortingSnapshot`, ()=>{
  it(`Should render Sorting`, () => {
    const store = mockStore({});

    const tree = renderer.create(
        <Provider store={store}>
          <Sorting
            activeSorting='popular'
            onSortItemClick={() => {}}
            isOpen={false}
            onToggleMenu={() => {}}
            onSelectMenu={() => {}}
          />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
