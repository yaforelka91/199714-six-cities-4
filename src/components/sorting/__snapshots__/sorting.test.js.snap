// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingSnapshot Should render Sorting 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  Â 
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom"
  >
    <li
      className="places__option places__option--active"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
  <select
    className="places__sorting-type"
    id="places-sorting"
    onChange={[Function]}
    style={
      Object {
        "border": 0,
        "clip": "rect(1px, 1px, 1px, 1px)",
        "height": "1px",
        "overflow": "hidden",
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": "1px",
      }
    }
    value="popular"
  >
    <option
      className="places__option"
      value="popular"
    >
      Popular
    </option>
    <option
      className="places__option"
      value="to-high"
    >
      Price: low to high
    </option>
    <option
      className="places__option"
      value="to-low"
    >
      Price: high to low
    </option>
    <option
      className="places__option"
      value="top-rated"
    >
      Top rated first
    </option>
  </select>
</form>
`;
